src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -293,11 +314,12 @@ public void validateArguments() {
 
     /**
      * Determine if a particular ReadFilter was disabled on the command line.
-     * @param filterName name of the filter to query
-     * @return true if the name appears in the list of disabled filters
+     * @param filterName name of the filter to query.
+     * @return {@code true} if the name appears in the list of disabled filters, or is a tool default not provided by the user and all tool defaults are disabled; {@code false} otherwise.
      */
     public boolean isDisabledFilter(final String filterName) {
-        return disableFilters.contains(filterName);
+        return disableFilters.contains(filterName)
+                || (disableToolDefaultReadFilters && !userReadFilterNames.contains(filterName));
```

Comment:
This is a fix for #2363. The new behavior of `--disableToolDefaultReadFilters` remove all the defaults filters only if the user one is not enable them again (if that change in `getAllInstances` is accepted). If the new behavior is removing all the default ones without allowing to re-enabled them, this line should be `(disableToolDefaultReadFilters && toolDefaultReadFilterNamesInOrder.contains(readFilter))`.

If someone is using this method to check if a required filter is disabled for a tool even if it is a default one, this should honor also the `--disableToolDefaultReadFilters`. Because the tool defaults were disabled in the command line if that parameter is provided, I think that it is still reflected in the first sentence. Anyway, I added in next commit a note saying that include disabled defaults if `--disableToolDefaultReadFilters` is provided in the command line.